### DISCUSSION TITLE
useRoutes() Path Regex

### DISCUSSION THREAD

**Discussion by goneglobal** [remix-run/react-router#8132] [Q&A] [answered]
Before in v4 I would do the following. 
It would only match if the path was /profile/<10 digit number> and provide the number as a param.

`<Route path="/profile/:userid([0-9]{10})" component={Profile} />`

Is there a way to regex url's like this in v6?
Or another solution to match like this.
If not I can put some logic in the component to handle it and navigate away if needed.
But I wanted to ask.

thank-you & V6 looks great!!

**Comment by timdorr** [comment, marked as answer]
No, because we no longer use the underlying `path-to-regexp` library. There are some thoughts about doing validation on paths, but nothing's been built yet.

**Reply by goneglobal** [reply]
no worries and thanks so much for replying.
Good luck with the development... thx

**Reply by RogerPodacter** [reply]
Hi! I'm new to the React Router and I was just wondering if there is a recommended approach or work around for achieving the functionality in @goneglobal's original post. Thanks!

**Reply by nick4fake** [reply]
Well, this seems like a big deal. We have upgraded everything to v6 just to notice that there is simply no way to replicate v5 functionality. Removing existing features without providing any alternative is a severe incompatibility.

**Reply by chris-mo** [reply]
Echoing @nick4fake sentiment here. This is a pretty big deal breaker for presumably more teams/devs than y'all think

**Reply by smff** [reply]
So is there a solution(pattern) yet, or is everyone out for themselves here?

upd: found answer at https://reactrouter.com/docs/en/v6/faq#what-happened-to-regexp-routes-paths

**Reply by sorsaffari** [reply]
This is a serious regression! I'm surprised this hasn't been prioritised, as many use-cases rely on this feature.

**Reply by Sean-Y-X** [reply]
Could you please add this feature back as it is very handy

**Reply by jnachtigall** [reply]
> upd: found answer at https://reactrouter.com/docs/en/v6/faq#what-happened-to-regexp-routes-paths

Gives a "404 Not found" now, corrected URL now is: https://reactrouter.com/docs/en/v6/getting-started/faq#what-happened-to-regexp-routes-paths

**Reply by jeffersoneagley** [reply]
What about lefthand-side wildcards? I have to catch _anything_ that ends with `*/auth`

**Reply by rodrigo-web-developer** [reply]
I know there is a problem with ranking conflict paths like /:param(first|second) and /:param([a-z]+), but if it is the only problem you can add an attribute "order" or "priority" to the Route component and let the developer to explicit sorts it. like:

```js
<Route path="/:param(first|second)" order={0} element={<FirstSecondComponent/>}/>
<Route path="/:param([a-z]+)" order={1} element={<AnyWordComponent/>}/>
```

There is no sense on validating params on component rendering and throws another component, it makes a bad designed component. And you will need to replicate this validation through the rest of components.

**Reply by finnharbeke** [reply]
> > upd: found answer at https://reactrouter.com/docs/en/v6/faq#what-happened-to-regexp-routes-paths
> 
> Gives a "404 Not found" now, corrected URL now is: https://reactrouter.com/docs/en/v6/getting-started/faq#what-happened-to-regexp-routes-paths

the current link is https://reactrouter.com/en/6/start/faq#what-happened-to-regexp-routes-paths, for those looking for it

**Reply by arcaela** [reply]
React Documentation move to [https://reactrouter.com/en/6.14.0/start/faq#what-happened-to-regexp-routes-paths](https://reactrouter.com/en/6.14.0/start/faq#what-happened-to-regexp-routes-paths)
